smsdk/client_v0.py

```diff
@@ -117,12 +119,26 @@ def __init__(self, tenant, site_domain="sightmachine.io"):
         :type site_domain: :class:`string`
         """
 
-        self.tenant = tenant
-
-        # Handle internal configuration
-        self.config = {}
-        self.config["protocol"] = "https"
-        self.config["site.domain"] = site_domain
+        if "://" in tenant:  # Full URL provided eg. "https://demo.sightmachine.io"
+            protocol, domain = tenant.split("://")[0], tenant.split("://")[1]
+            self.tenant = domain.split(".")[0]
+            self.config = {
+                "protocol": protocol,
+                "site.domain": ".".join(domain.split(".")[-2:]),
+            }
+        else:
+            tenant_parts = tenant.split(".")
+            if (
+                len(tenant_parts) > 1
+            ):  # Tenant with domain provided but missing protocol eg. "demo.sightmachine.io"
+                self.tenant = tenant_parts[0]
+                self.config = {
+                    "protocol": protocol,
+                    "site.domain": ".".join(tenant_parts[1:]),
+                }
+            else:  # Only tenant name provided eg. "demo"
+                self.tenant = tenant
+                self.config = {"protocol": protocol, "site.domain": site_domain}
```

Comment:
Might be better to normalize the parameters in the condition and then setup the structure:
```
   if X
     tenant = tenant
     protocol = s
     site_domain = sd
   elif Y:

   # Else, leave parameters alone.  Or validate them

   self.tenant = tenant
   self.config = {"protocol": protocol, "site.domain": site_domain}
```

Reply:
Done.